central/sensor/service/connection/connection_test.go

```diff
@@ -57,6 +87,63 @@ func (c *mockServer) Recv() (*central.MsgFromSensor, error) {
 	return nil, nil
 }
 
+func (s *testSuite) TestSendsScanConfigurationMsgOnRun() {
+
+	ctx := sac.WithAllAccess(context.Background())
+
+	ctrl := gomock.NewController(s.T())
+	mgrMock := clusterMgrMock.NewMockClusterManager(ctrl)
+	pipeline := pipelineMock.NewMockClusterPipeline(ctrl)
+	deduper := mocks.NewMockDeduper(ctrl)
+	stopSig := concurrency.NewErrorSignal()
+
+	hello := &central.SensorHello{
+		SensorVersion: "1.0",
+	}
+
+	eventHandler := newSensorEventHandler(&storage.Cluster{}, "", pipeline, nil, &stopSig, deduper, nil)
+
+	sensorMockConn := &sensorConnection{
+		//clusterID:          fixtureconsts.Cluster1,
+		clusterMgr:         mgrMock,
+		sensorEventHandler: eventHandler,
+		sensorHello:        hello,
+		hashDeduper:        deduper,
+		scanSettingDS:      s.scanConfigDS,
+		sendC:              make(chan *central.MsgToSensor),
+	}
+
+	caps := set.NewSet[centralsensor.SensorCapability]([]centralsensor.SensorCapability{centralsensor.SendDeduperStateOnReconnect}...)
+
+	mgrMock.EXPECT().GetCluster(ctx, gomock.Any()).Return(&storage.Cluster{}, true, nil).AnyTimes()
```

Comment:
I think this should be called only twice.

Reply:
I can do that, but then this test will have to be touched again if the number of GetCluster calls changes, I didn't think it mattered for this test how often that was called.